src/lld/moviebooking/class_diagram.puml

```diff
@@ -1,3 +1,323 @@
 @startuml
+'Main Entity and Relational mapping entity
 
+package com.movie_booking.entity.main{
+    class Identifier{
+        - id: long
+        - name: String
+    }
+    class User extends Identifier{
+        - email: String
+        - phoneNo: String
+    }
+    class Movie extends Identifier{
+        - crew: List<Crew>
+        - supportedLanguages: List<LanguageCode>
+    }
+    class Crew extends Identifier{
+        - crewType: CrewType
+    }
+    enum CrewType{
+        Cameraman
+        Designer
+        Hero
+        Heroine
+        Director
+        Singer
+    }
+    'Todo: Preferred hierarchical approach, should discuss and final best entity design for city state, this takes four level joins
+    class District extends Identifier{
+        cityId: long
+    }
+    class City extends Identifier{
+        stateId: long
+    }
+    class State extends Identifier{
+        countryId: Country
+    }
+    class Country extends Identifier{
```

Comment:
should each of this address classification have thier respective Ids along with their parent group.
For example City should be 
city{
cityId,
stateId
}

Reply:
I believe  abstract base class Identifier holds it cityId, Am i missing anything here?
``` class City extends Identifier{
        stateId: long
    }